GLMakie/src/GLAbstraction/GLTypes.jl

```diff
@@ -345,6 +365,11 @@ function RenderObject(
 
     switch_context!(context)
 
+    # This is a lazy workaround for disabling updates of `requires_update` when
+    # not rendering on demand. A cleaner implementation should probably go 
+    # through @gen_defaults! and adjust constructors instead.
+    track_updates = to_value(pop!(data, :track_updates, true))
```

Comment:
`@gen_defaults` is pretty much deprecated (without a new implementation yet, but I think it's pretty clear that it's pretty redundant), so I guess that's fair ;)

Reply:
The way it works atm is that GPUArrays always attach the update trackers when they get created. When the renderobject is created it either removes the tracking (i.e. observer functions) or connects the remaining uniforms based on what would usually be interpreted as a uniform. Seems quite hacky to me :shrug: 

A clean implementation wouldn't connect (and maybe also not define) the tracking observables in GPUArrays. I think with how things are organized atm `@gen_defaults` would need to be adjusted to forward that information. But I don't really want to mess with that macro. I'd probably just break it

Reply:
I meant, `@gen_defaults` needs to be refactored / removed going forward, so whenever we do that, we can clean up the implementation of `track_updates` ;)

Reply:
In other words: please don't waste a second on `@gen_defaults` ;)

Reply:
Ok I think the pr is ready then